src/boot/efi/stub.c

```diff
@@ -110,23 +110,26 @@ static VOID export_variables(EFI_LOADED_IMAGE *loaded_image) {
                 _cleanup_freepool_ CHAR16 *s = NULL;
 
                 s = DevicePathToStr(loaded_image->FilePath);
-                efivar_set(LOADER_GUID, L"LoaderImageIdentifier", s, 0);
+                if (s)
+                        efivar_set(LOADER_GUID, L"LoaderImageIdentifier", s, 0);
```

Comment:
i think the idea here was that we invalidate whatever is set before in case of failure. efivar_set() with a NULL param after all removes the existing variable. Quite frnakly, i really don't like to do things that way though, it's too magic for my test. At least should have a comment... but we should probably log about the error here (even if we proceed)